pkg/api/message/subscribe_worker.go

```diff
@@ -178,9 +178,17 @@ func startSubscribeWorker(
 			nodeID := uint32(env.OriginatorNodeID)
 			seqID := uint64(env.OriginatorSequenceID)
 
-			if current, ok := lastSeen[nodeID]; !ok || seqID > current {
-				lastSeen[nodeID] = seqID
+			if last, ok := lastSeen[nodeID]; ok && seqID < last {
```

Comment:
I do worry about abuse potential here.  
  
Node A can publish bad payloads and completely take down Node B. It's fine for testing but I worry about this in prod.  
  
A middle-ground would be completely shutting down the stream to the node that breaks this invariant, and we fire a metric that alerts on the first call.

Reply:
Oh, I’m sorry. I thought this was on the read from the stream. Agree these never should get into the DB in the first place.